### ISSUE TITLE
Many admin pages do not have the new look 

### ISSUE THREAD

**Issue by scottcwilson** [zencart/zencart#996]
Admin home page looks good, as do LEAD reports, but admin-> configuration-> anything and admin->customers->customers look bad - pale header bar, hard to read links on the right.

**Comment by zcwilt** [comment]
The ultimate intention is for all admin page to be converted to the new Controller logic, 
However in the mean time some bridging code is probably needed

**Comment by scottcwilson** [comment]
I wonder if we can get some css wizard to style the old pages to look like the new ones as a baby step towards getting them more aligned.  Once I successfully do the Find Duplicate Models report, I'll work on some of the other reports, which are all using the old code at the moment.

**Comment by zcwilt** [comment]
Unfortunately it's not quite as simple as that.

Pages that have been converted to the new controller logic, use the $tplVars array to pass information to the view logic.

legacy pages don;t set any $tplVars, so view that need them will break ( e.g the new adminlte special menu)

That problem isn't insurmountable, but is a code fix rather than a css fix.

**Comment by dakanji** [comment]
I'll like to take this on (at least to make the look of the pages be consistent as a start by moving them to bootstrap).
Actually, I had started the process by setting up a structured process here: https://github.com/dakanji/zencart-style which I was going to discuss with @zcwilt, @drbyte, @ajeh to add to the project for generating the CSS and JS files in a structured.
I can move this over to the template being used on the index page and then start to roll this across to the other pages. 

The template implementation currently has developer only files files in it that should not be distributed and a few other small issues (should be in a stand alone repo and only the outputs from the dist folder should be distributed).

From the way the template has been deployed, perhaps the dev team is not that up to speed with task runners such as Gulp/Grunt or CSS extension languages such Sass/Less but it is a cakewalk once the dev environment is set up.

Bootstrap and the template use Less but Less has basically lost out to Sass. Bootstrap 4 will be Sass based and there is already an official Sass port of Bootstrap 3. So, I will set this up using Sass (which incidentally can take pure CSS code directly so even if not fully understood, can be updated with CSS directly.

If okay with you, I'll get on moving the candidate repo over to generating the template files and we can talk through things as we progress.

**Comment by drbyte** [comment]
@dakanji 

Undecided about where you're headed here.

This issue ticket is about Admin things, but the way you worded things suggests you're aiming at the entire system. Is that an accurate interpretation?


Not sure I want a million additional dependencies added anywhere. Open to `gulp`. Not sure we want `Handlebars` dependencies. I'd be more likely to use `vuejs` if we're going to get into js templating et al.

I agree with your point about less/sass: `sass` is preferable. Forget `less`.


--

NOTE: in the previous PR where I asked you to "capitalize Zen Cart" (you were using lowercase one word 'zencart'), you interpreted that to mean "ZENCART". I meant literally: "Zen Cart" (ie: capital Z and capital C, 2 separate words). In that other PR I didn't bother getting petty about it. But since you're doing more things, it's best to use the name correctly: 2 words, each in Proper Case.

**Comment by dakanji** [comment]
On the product name, I did think the all caps seemed strange but it didn't cross my mind to verify but instead, I just assumed you were probably a bit loopy and decided not to be petty about it :) 

On the proposal, this is aimed at the Admin side ... It could of course be extended to the catalogue at some stage if needed but that is not the target and hasn't even been considered at this stage ... but technically possible for the default themes. I will pipe the outputs into into an "Admin" folder so that other outputs can be directed to a "Catalogue" folder at some point down the line if required.

The handlebars thing is a bit of a red herring as I started off from a private setup where it is used to generate static HTML files on a completely different project. In my "zencart-style" repo, it is currently only used to generate a style guide for the previous Bootswatch based implementation and is of no concern to an end user as it doesn't contribute to the actual files intended for use. 

In other words, it shouldn't be in the README as there is no intention to get into either one of html or js templating. Will get to fixing up the README and removing the implementation later as still setting things up, converting from less to sass, trying to see which of the many js plugins and associated css distributed with AdminLTE, some of which are alternatives to each other, are being used and how best to deploy/package them and such at this point.

My proposal is to:

- Set up a structured way to generate, and update, key JS and CSS assets
- Set up a structured way to present these assets, such as auto applying/removing browser prefixes, standardising presentation format and order, concatenation, minification and the several other things you can do with a task runner such as Gulp.
- Update the implementation of these in the Zen Cart admin. That is, delete developer only files not meant for distribution etc and update the output files. Immediate targets are the legacy files such that when the coding updates mentioned are implemented, there is already the look/feel in place.

The idea is that there is a separate repo for the first two items (happy to set up and let the team adopt and modify to taste). This will spit out files that can then be dropped into the appropriate place(s) in the main repo. This can be automated as a separate task in Gulp for instance but I envisage this being done manually as a start. Aim is to get the basics in place and then look at optimisation later.

**Comment by drbyte** [comment]
I'd be willing to look at a prototype. Admittedly there's some resistance, which I'm trying to navigate.

Also worth noting: We're not 10000% committed to AdminLTE, as it feels bloated for what we really need. It has a few potentially useful plugins, but some of them are dated. And it appears AdminLTE is not currently 'actively maintained'.

**Comment by dakanji** [comment]
I can understand resistance as I would have been so inclined a year or so ago before trying NodeJS out.

AdminLTE is bloated when simply deployed as distributed. We need to bear in mind when dealing with such templates that there is large showcase element with multiple options and that a person deploying has to cherry pick what is actually needed from the set.

The beauty of using a task runner is that the whole bloated shebang can be dumped into the repo and the needed items cherry picked using Gulp. If or when another item is needed, edit the Gulpfile to add it, run this and you are good to go.

In terms of a prototype, there is already one so to speak. Just needs the output folder structure in the updated to match the target project structure.

If you have a NodeJS environment in place, you can clone the repo, install the dependencies (npm install && bower install), checkout the bootswatch branch and run "npm run dev" to get a style guide which will be loaded in your default browser. 

The style guide is a "dark" theme based on the [Bootswatch Slate Theme](https://bootswatch.com/slate/). There is also a "light" theme based on the [Bootswatch Paper Theme](https://bootswatch.com/paper/) with a modified colour scheme. To get this, edit change "/dark/" to "/light/" in the three sass files directly under "src/assets/scss". 

The light theme is actually the more stable one and was porting it over to the dark theme, as I thought it might look nicer, before I noticed the recent PRs but it should be working okay. 

The assets (with sourcemaps for debugging the JS/CSS in a browser) will be in a "dist/dev" folder that will be created. 

For production/deployment, "npm run prod" will create minified files in "dist/production" and clear text sources (same as used for dev runs) in "dist/sources".

The AdminLTE branch will be set up to do similar.

Main thing is to get the NodeJS dev environment setup which should not be a big issue and bear in mind that this is not targeted at the regular user. 

To setup NodeJS on a Mac, I suggest following [this guide](http://blog.teamtreehouse.com/install-node-js-npm-mac). Using homebrew instead of the package installer distributed by the NodeJS team for this really makes things straightforward.

Once this is in place, a whole new world is opened.

**Comment by drbyte** [comment]
I grok the value of task runners and automated preprocessing. 
(I use gulp regularly in other work, particularly with https://laravel.com/docs/master/elixir / https://www.sitepoint.com/meet-elixir-the-laravel-way-of-compiling-assets/ )
... and I'd be inclined to lean that direction, if only for the fact that I'm tired and don't want to have to document yet another way to do things ;)

**Comment by dakanji** [comment]
Cool but not sure what the benefit of installing a PHP framework just to run Gulp is. Suppose it might appeal to someone that already uses Lavarel. I personally don't and would rather not if possible although I might be biased against PHP frameworks after a project I used to contribute to went belly up after dumping perfectly good code to adopt the now moribund Kohana

Firstly anyway, looking at the plugins bundled with AdminLTE, I notice a few of them are under the Apache 2 license which I understand means they cannot be be distributed with ZC being under the GPL v2 license (The usual open source licensing nightmare). Is this correct and they have to be removed?

**Comment by scottcwilson** [comment]
>  a project I used to contribute to went belly up after dumping perfectly good code 
> to adopt the now moribund Kohana
Wait - this sounds like a move *I* would make!  LOL.

**Comment by drbyte** [comment]
Elixir can be installed on its own. In essence it's not a lot different than what you're proposing. Just two different approaches. Flavours of the month. This whole "space" is rapidly changing. 

AdminLTE itself is MIT. Perhaps some plugins won't be suitable.

**Comment by dakanji** [comment]
@scottcwilson: They didn't adopt a moribund framework. it was flavour of the month along with cakephp a few years ago which was when they decided to jump on board and throw out everything done before ... years of hard work on beautiful code.

@drbyte: Had a look at Elixir despite trying to get over Brexit 2.0 at the hands of Iceland. Seems to be an abstraction layer over Gulp. I will prefer to proceed without learning an abstraction API to do what can easily be done directly and has already been setup.

If the team later wishes to impose a Elixr abstraction over this, then this can be done but I would suggest it is better to just run "Gulp this" and "Gulp that" directly.

I also think it would be better for others down the line as Gulp can be used directly even if they have the Elixir setup but if based on Elixir, users will be forced to implement an Elixir setup.

Let's just say I am not a big fan of abstraction layers particularly for small, short and sharp tasks but if there is a bigger picture I am missing, I'll suck it, grit my teeth and get on with it.

**Comment by drbyte** [comment]
NOTE: I've removed a plethora of unused things that came with the AdminLTE stuff.

**Comment by dakanji** [comment]
Noted.

**Comment by dakanji** [comment]
Sorry for break in transmission.

@drbyte, there is a prototype in place. Looking forward to feedback.
Will get down to doing the styling stuff on this is the meantime. Will probably break this down into chunks.

**Comment by zcwilt** [comment]
Just a note. If some one else wants to use the same process as I have used to convert other admin pages, please do a few pages at a time. Huge commits that hit 20-30+ files will impinge on others development branches.

**Comment by dakanji** [comment]
True. Will be proposing the changes in a structured page by page manner (left to right on the admin menu) but first with the login and password reset pages which could do with a facelift. 

Just need to finalise the Gulp/Bootstrap usage part which I should be wrapping up over this weekend.  I have finished this for an AdminLTE option but think AdminLTE may not be the best option to take forward and tidying up an alternative Bootswatch option to put forward for consideration which I think will be a lot more easy to use.

WIll open up discussion on this later to seek the team's opinions/preferences.

**Comment by dakanji** [comment]
This hasn't been formally assigned to me but as I did make a long pitch about it, just wanted to flag that, due to other pressing issues, I can't take this on for the foreseeable future just in case someone wants to take it on and was sitting back because of my pitch.

The mentioned prototype is here: https://github.com/dakanji/zencart-style in case there is interest but probably best to follow another path on that aspect.
